delivery-metrics/src/delivery_metrics_deliverable_model.py

```diff
@@ -0,0 +1,38 @@
+from delivery_metrics_model import DeliveryMetricsModel
+
+class DeliveryMetricsDeliverableModel(DeliveryMetricsModel):
+
+	def syncDeliverable(self, deliverable: dict) -> int:
+		
+		# validation
+		if not isinstance(deliverable, dict):
+			return None
+
+		# get values needed for sql statement
+		guid = deliverable.get('guid')
+		title = deliverable.get('title')
+		pillar = deliverable.get('pillar')
+		quad_id = deliverable.get('quad_id')
+		effective = self.getEffectiveDate()
```

Comment:
I like the reusable pattern that you landed on for sharing the logic around the model and inserts/updates!

We should talk about this further as we look toward migrating this to the `HHS/simpler-grants-gov` repo, but a fairly common pattern with python is to use the [SQLAlchemy ORM](https://docs.sqlalchemy.org/en/20/orm/) to manage the "model" and then inject the model as a dependency into a separate service that handles the reads/writes. That's roughly the pattern followed by the API codebase.

I know ORMs can be a highly debated topic but, there are some broader pros with using SQLAlchemy in that it helps you abstract the specific database you're using to a simple connection string without having to re-write any SQL to align with a specific dialect. Having the models defined in the code also makes spinning up a mock db really easy for testing.

But for the prototype this great! And I appreciate you didn't reach for any external dependencies.

Reply:
Thanks! Let's explore ORM options in subsequent iterations.